eng/scripts/distribute-packages-to-matrix.ps1

```diff
@@ -0,0 +1,263 @@
+<#
+.SYNOPSIS
+Distributes a set of packages to a platform matrix file by adding computing and adding a TargetingSTring property
+dynamically. If an unexpected situation is encountered, the script will make no changes to the input file.
+
+.DESCRIPTION
+Because of the way the platform matrix file is structured, we need to distribute the packages in a way that
+honors the Include packages. We have these included this way because want to ensure that these python versions
+run on that platform. This script is aware of these additional configurations and will set TargetingString for
+them as well.
+#>
+param(
+    [parameter(Mandatory=$true)]
+    [string]$PackageInfoFolder,
+    [parameter(Mandatory=$true)]
+    [string]$PlatformMatrix
+)
+
+$RepoRoot = (Resolve-Path (Join-Path $PSScriptRoot ".." ".."))
+Set-StrictMode -Version 4
+$BATCHSIZE = 8
+
+if (!(Test-Path $PackageInfoFolder)) {
+    Write-Error "PackageInfo folder file not found: $PackageInfoFolder"
+    exit 1
+}
+
+if (!(Test-Path $PlatformMatrix)) {
+    Write-Error "Platform matrix file not found: $PlatformMatrix"
+    exit 1
+}
+
+function Extract-MatrixMultiplier {
+    param (
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject]$Matrix
+    )
+
+    $highestCount = 1
+
+    foreach ($property in $Matrix.PSObject.Properties) {
+        $type = $property.Value.GetType().Name
+        switch ($type) {
+            "PSCustomObject" {
+                $itemCount = 0
+
+                # this looks very strange to loop over the properties of a PSCustomObject,
+                # but this is the only way to actually count. The Count/Length property is
+                # NOT available
+                foreach($innerProperty in $property.Value.PSObject.Properties) {
+                    $itemCount++
+                }
+
+                if ($itemCount -gt $highestCount) {
+                    $highestCount = $itemCount
+                }
+            }
+            "Object[]" {
+                $count = $property.Value.Length
+                if ($count -gt $highestCount) {
+                    $highestCount = $count
+                }
+            }
+        }
+    }
+
+    return $highestCount
+}
+
+function Split-ArrayIntoBatches {
```

Comment:
I have a balanced array function in another matrix generating script in eng/common:
https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/New-RegenerateMatrix.ps1#L25

This could be extracted into a helper method for use in other scripts.

Reply:
I'm not doing that for a PR that's already lasted 2 months. This needs to MOVE.

Reply:
I already had an immediate follow-up of migrating the combo in build-package-artifacts to your abstraction of save-package-properties. 

In the same PR that updates that I'll make use of that other function.